### ISSUE TITLE
No mask generated by model.detect

### ISSUE THREAD

**Issue by suchiz** [matterport/Mask_RCNN#1836]
Hello everyone,

I've trained MaskRCNN with my own dataset. It runs well, but it doesn't generate masks. The BBOX are really great and accurate, the classification aswell but no mask generated.

I am based on the balloon.py to generate my mask from VIA annotator.
I decided to take a deeper look in "load_mask", and when I plot the masks generated, it works well. I have the polygon annotated from VIA in white , and the rest in black. Yet, all the mask returned by model.detect are full of zeros... Once again, the BBOX and the classification are perfect !

If you have any guideline to help me out, it will be really great. Thank you !

PS: While someone reply to this, i'll save all the mask generated to a folder per image like a classical dataset with masks given ...

**UPDATE:**
Little update, it turns out that in the function unmold_mask from utils.py, my mask values are all under the threshold value which 0.5... but I don't know why yet.

**Comment by suchiz** [comment]
There was no problem with the mask generation of balloon.py, the problem was the deprecated scipy.misc.imresize function. I was replacing this function with PIL.Image.resize() of Pillow + Numpy Array but it wasn't returning the same thing, not the same values. After downgrading scipy, it was working well !

**original:** 
        m = scipy.misc.imresize(m.astype(float), mini_shape, interp='bilinear')

**what I've replaced, but not working:**
        #im = Image.fromarray(m.astype(float))
        #m = np.array(im.resize(mini_shape, PIL.Image.BILINEAR))

**Comment by yhc1994** [comment]
Hi suchiz, could you share me your scipy version? I met the same issue that the model.detect cannot return the correct mask

**Comment by suchiz** [comment]
Hi there. Any version below 1.3.0. currently using 1.2.1 but got a warning which says it is deprecated.

**Comment by yhc1994** [comment]
> Hi there. Any version below 1.3.0. currently using 1.2.1 but got a warning which says it is deprecated.

Unfortunately there is still some mistakes of mask after I download scipy to 1.2.1, the BBOX is located very accurately but the mask will not exist in some of BBOXs

**Comment by suchiz** [comment]
Unfortunately I cannot help you more with this little information... Did you check the mask generated in 'load_mask' function ? I mean just plot it to check if everything was fine ? If you have enough memory, you can try to run it without 'mini mask' parameter if it is really the same problem as I had (resize). But it looks like it isn't

**Comment by NizarTarabay** [comment]
> > Hi there. Any version below 1.3.0. currently using 1.2.1 but got a warning which says it is deprecated.
> 
> Unfortunately there is still some mistakes of mask after I download scipy to 1.2.1, the BBOX is located very accurately but the mask will not exist in some of BBOXs

Hello @yhc1994 
I'm facing the same issue, the BBs are well detected but the network fails to detect their masks.
Did you manage to solve this issue?
Thank you!

**Comment by czHappy** [comment]
Thanks, it works well. I downgrade the version of scipy to 1.2.1.  My environment configuration is as follows:
python: 3.6
tensorflow: 1.12.0
keras: 2.2.4
